src/api_client.rs

```diff
@@ -89,297 +95,301 @@ impl Client {
-    }
-
-    pub async fn get_node_version() -> Result<String, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_sync_status() -> Result<SyncStatus, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_health() -> Result<HealthStatus, Error> {
-        unimplemented!("")
-    }
-
-    /* validator namespace */
-    pub async fn get_attester_duties(
-        epoch: Epoch,
-        indices: &[ValidatorIndex],
-    ) -> Result<(Root, Vec<AttestationDuty>), Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_proposer_duties(epoch: Epoch) -> Result<(Root, Vec<ProposerDuty>), Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_sync_committee_duties(
-        epoch: Epoch,
-        indices: &[ValidatorIndex],
-    ) -> Result<(Root, Vec<SyncCommitteeDuty>), Error> {
-        unimplemented!("")
-    }
-
-    // v2 endpoint
-    pub async fn get_block(
-        slot: Slot,
-        randao_reveal: RandaoReveal,
-        graffiti: Bytes32,
-    ) -> Result<BeaconBlock, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_blinded_block(
-        slot: Slot,
-        randao_reveal: RandaoReveal,
-        graffiti: Bytes32,
-    ) -> Result<BlindedBeaconBlock, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_attestation_data(
-        slot: Slot,
-        committee_index: CommitteeIndex,
-    ) -> Result<AttestationData, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn get_attestation_aggregate(
-        attestation_data_root: Root,
-        slot: Slot,
-    ) -> Result<Attestation, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn post_aggregates_with_proofs(
-        aggregates_with_proofs: &[SignedAggregateAndProof],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
-
-    pub async fn subscribe_subnets_for_attestation(
-        committee_descriptors: &[CommitteeDescriptor],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
-
-    pub async fn subscribe_subnets_for_sync_committee(
-        sync_committee_descriptors: &[SyncCommitteeDescriptor],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
-
-    pub async fn get_sync_committee_contribution(
-        slot: Slot,
-        subcommittee_index: usize,
-        beacon_block_root: Root,
-    ) -> Result<SyncCommitteeContribution, Error> {
-        unimplemented!("")
-    }
-
-    pub async fn post_contributions_with_proofs(
-        contributions_with_proofs: &[SignedContributionAndProof],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
-
-    pub async fn prepare_proposers(
-        registrations: &[BeaconProposerRegistration],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
-
-    // endpoint for builder registrations
-    pub async fn register_validators_with_builders(
-        registrations: &[SignedValidatorRegistration],
-    ) -> Result<(), Error> {
-        Ok(())
-    }
+    // pub async fn get_genesis_details(&self) -> Result<GenesisDetails, Error> {
```

Comment:
you shouldn't need to comment out everything here and it makes it harder to review the actual changes bc we end up with a large diff of just comments